Mytinerary/View Controllers/Details View Controllers/ItineraryDetailsViewController.m

```diff
@@ -87,15 +97,51 @@ - (void)reloadEventTable {
     }];
 }
 
-/*
-#pragma mark - Navigation
+#pragma mark - Button Functions
 
-// In a storyboard-based application, you will often want to do a little preparation before navigation
-- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
-    // Get the new view controller using [segue destinationViewController].
-    // Pass the selected object to the new view controller.
+- (IBAction)didTapDeleteItinerary:(id)sender {
+    
+    // display an alert asking for confirmation of deletion
+    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Delete Itinerary?" message:@"Are you sure you want to delete this itinerary. This will delete all this itinerary's events. This cannot be undone." preferredStyle:UIAlertControllerStyleAlert];
+
+    // create Delete and Cancel buttons to alert
+    UIAlertAction* deleteAction = [UIAlertAction actionWithTitle:@"Delete" style:UIAlertActionStyleDefault
+    handler:^(UIAlertAction * action) {
+        // if the delete button is pressed again
+        if (action) {
+            // delete the itinerary's events individually
+            for (Event *eventToBeDeleted in self.itinerary.events) {
+                [eventToBeDeleted deleteInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {}];
+            }
+            
+            [self.itinerary deleteInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
```

Comment:
can we factor some of this out to the itinerary class? we can make a "deleteItinerary" function that takes in a completion block